### PR TITLE
Feat/environment strategies

### PR DESCRIPTION
**Pull request by FredrikOseberg**

### CODE REVIEW

File: src/component/feature/FeatureView2/FeatureStrategies/FeatureStrategiesEnvironments/FeatureStrategiesEnvironmentList/useDeleteStrategyMarkup.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+            title="Are you sure you want to delete this strategy?"
+            open={show}
+            primaryButtonText="Delete strategy"
+            secondaryButtonText="Cancel"
+            onClick={onClick}
+            onClose={onClose}
+        >
+            <Alert severity="error">
+                Deleting the strategy will affect which users receive access to
+                the feature.
```
Comment (chriswk) [inline src/component/feature/FeatureView2/FeatureStrategies/FeatureStrategiesEnvironments/FeatureStrategiesEnvironmentList/useDeleteStrategyMarkup.tsx:26]:
```suggestion
                Deleting the strategy will change which users receive access to
                the feature.
```

File: src/component/feature/FeatureView2/FeatureStrategies/FeatureStrategiesEnvironments/FeatureStrategiesRefresh/FeatureStrategiesRefresh.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            leave={styles.fadeInLeave}
+        >
+            <div className={styles.container}>
+                <p className={styles.refreshHeader}>
+                    NOTE: Updated configuration
+                </p>
+                <p className={styles.paragraph}>
+                    There is new strategy configuration available. This might
+                    mean that someone has updated the strategy configuration
+                    while you were working. Would you like to refresh?
```
Comment (chriswk) [inline src/component/feature/FeatureView2/FeatureStrategies/FeatureStrategiesEnvironments/FeatureStrategiesRefresh/FeatureStrategiesRefresh.tsx:32]: This is really cool.

Next; we should figure out how to make strategies CRDTs so we can support simultaneous edits :)

File: src/component/feature/FeatureView2/FeatureStrategies/FeatureStrategyExecution/FeatureStrategyExecution.tsx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+                <FeatureStrategyExecutionConstraint
+                    constraint={constraint}
+                    key={`${constraint.contextName}-${index}`}
+                />
+            );
+        });
+    };
+
+    const renderParameters = () => {
+        console.log(parameters);
```
Comment (chriswk) [inline src/component/feature/FeatureView2/FeatureStrategies/FeatureStrategyExecution/FeatureStrategyExecution.tsx:45]:
```suggestion
```

File: src/component/feature/strategy/EditStrategyModal/general-strategy.jsx [hunk 4]
```diff
@@ -22,104 +32,122 @@ export default function GeneralStrategyInput({ parameters, strategyDefinition, u
         const value = currentValue === 'true' ? 'false' : 'true';
         updateParameter(key, value);
     };
+    console.log(strategyDefinition);
```
Comment (chriswk) [inline src/component/feature/strategy/EditStrategyModal/general-strategy.jsx:35]:
```suggestion
```

File: src/component/menu/routes.js [hunk 5]
```diff
@@ -242,6 +243,16 @@ export const routes = [
         layout: 'main',
         menu: {},
     },
+    {
```
Comment (chriswk) [inline src/component/menu/routes.js:246]: I guess we plan to replace/rename these routes to just features once we're satisfied we've landed the correct view?

Comment (chriswk) [reply src/component/menu/routes.js:246]: Instead of `features2` that is

File: src/hooks/useToast.tsx [hunk 6]
```diff
@@ -1,15 +1,21 @@
...
 
+export interface IToast {
+    show: boolean;
+    type: string;
+    text: string;
+}
+
 const useToast = () => {
-    const [toastData, setToastData] = useState({
+    const [toastData, setToastData] = useState<IToast>({
```
Comment (chriswk) [inline src/hooks/useToast.tsx:11]: Cool

**Review by chriswk — APPROVED** [review]
I'm loving the hook/getters pattern using SWR instead of redux, a lot easier to wrap my head around.

I'm not seeing anything stopping us from merging, particularly since it's on a new path.

Some questions and nits included as comments added inline to the code , but I'd say go for it.

File: src/component/feature/FeatureView2/FeatureStrategies/FeatureStrategiesEnvironments/FeatureStrategiesEnvironmentList/useDeleteStrategyMarkup.tsx (hunk 1 above)
Comment (chriswk) [reply src/component/feature/FeatureView2/FeatureStrategies/FeatureStrategiesEnvironments/FeatureStrategiesEnvironmentList/useDeleteStrategyMarkup.tsx:26]:
```suggestion
                Deleting the strategy may change which users receive access to
                the feature.
```

**Review by ivarconr — APPROVED** [review]

**Pull request merged**